### ISSUE TITLE
segment max does not output zero for empty segment

### ISSUE THREAD

**Issue by fuzzyswan** [tensorflow/tensorflow#58123]
### Issue Type

Bug

### Source

binary

### Tensorflow Version

tf 2.10.0

### Custom Code

Yes

### OS Platform and Distribution

Linux Ubuntu 20.04

### Current Behaviour?

```shell
According to document, if the max is empty for a given segment ID i, output[i] = 0. However, on gpu it gives a large negative value for empty segment ID.
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf

data = np.array([[1, 2, 3, 4], [5, 6, 7, 8], [4, 3, 2, 1], [8, 7, 6, 5]])
segment_ids = [1, 2, 3, 4]
segment_ids = tf.constant(segment_ids, dtype=tf.int32)
data = tf.constant(data, dtype=tf.float32)
with tf.device('gpu'):
    output = tf.raw_ops.SegmentMax(data=data, segment_ids=segment_ids)
    print(output.numpy())
    output = tf.math.segment_max(data=data, segment_ids=segment_ids)
    print(output.numpy())
with tf.device('cpu'):
    output = tf.raw_ops.SegmentMax(data=data, segment_ids=segment_ids)
    print(output.numpy())
    output = tf.math.segment_max(data=data, segment_ids=segment_ids)
    print(output.numpy())
```

### Relevant log output

```shell
[[-3.4028235e+38 -3.4028235e+38 -3.4028235e+38 -3.4028235e+38]
 [ 1.0000000e+00  2.0000000e+00  3.0000000e+00  4.0000000e+00]
 [ 5.0000000e+00  6.0000000e+00  7.0000000e+00  8.0000000e+00]
 [ 4.0000000e+00  3.0000000e+00  2.0000000e+00  1.0000000e+00]
 [ 8.0000000e+00  7.0000000e+00  6.0000000e+00  5.0000000e+00]]
[[-3.4028235e+38 -3.4028235e+38 -3.4028235e+38 -3.4028235e+38]
 [ 1.0000000e+00  2.0000000e+00  3.0000000e+00  4.0000000e+00]
 [ 5.0000000e+00  6.0000000e+00  7.0000000e+00  8.0000000e+00]
 [ 4.0000000e+00  3.0000000e+00  2.0000000e+00  1.0000000e+00]
 [ 8.0000000e+00  7.0000000e+00  6.0000000e+00  5.0000000e+00]]
[[0. 0. 0. 0.]
 [1. 2. 3. 4.]
 [5. 6. 7. 8.]
 [4. 3. 2. 1.]
 [8. 7. 6. 5.]]
[[0. 0. 0. 0.]
 [1. 2. 3. 4.]
 [5. 6. 7. 8.]
 [4. 3. 2. 1.]
 [8. 7. 6. 5.]]
```

**Comment by mohantym** [comment]
Hi @fuzzyswan !
I got same results after specifying the gpu cards. It is always suggested to specify Gpu cards in tf.device() api.

Attached [gist](https://colab.sandbox.google.com/gist/mohantym/58c8f248295860a289b4bbe5499d9e10/git_58123.ipynb#scrollTo=PF2085DW7YME) for reference.

Thank you!

**Comment by SuryanarayanaY** [comment]
@fuzzyswan, I have replicated the reported behaviour on a Ubuntu VM. To be more specific the reported problem observed when we used `dtype` as `float` and with `dtype` as `int` there is no difference in CPU vs GPU output.

Result with data `dtype = float32`(Different on both CPU & GPU):

<img width="889" alt="Screenshot 2023-02-11 at 3 58 36 PM" src="https://user-images.githubusercontent.com/116063290/218406528-9a684a5c-f571-44db-810c-9ac9d378f41d.png">

Result with data `dtype = int32`(same on both CPU & GPU) :

<img width="1487" alt="Screenshot 2023-02-11 at 4 02 37 PM" src="https://user-images.githubusercontent.com/116063290/218406461-4f8423fe-bf06-427f-b9e4-58560558d7cb.png">

So the GPU outputs Zero as very large Negative number for the APIs `tf.raw_ops.SegmentMax` as well as `tf.math.segment_max` when dtype is `float`. This needs investigation as the difference between results is very huge. Thanks!
